torch_xla/csrc/ops/infer_output_shape.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "tensorflow/compiler/xla/client/xla_builder.h"
+#include "tensorflow/core/lib/gtl/array_slice.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+
+using LowerForShapeFn = std::function<xla::XlaOp(
+    tensorflow::gtl::ArraySlice<const xla::XlaOp> operands)>;
+
+// Compute the output shape for the given input shapes and lowering.
+xla::Shape InferOutputShape(
+    tensorflow::gtl::ArraySlice<const xla::Shape> input_shapes,
```

Comment:
I'd take const Shape* here.
Shape is heavy, and is becoming heavier.
Maybe.
I don't know.
The XLA builder is not light either.

Reply:
That'd make the caller a mess, let's not do it until we know it's a problem. Also, the slice shouldn't copy the elements, so it's not *that* bad.

Reply:
It does copy. The std::initializer_list<> does.
Wouldn't be a big mess. Just prepend an & in front.
If you want to leave it as is I am OK.
But it's going to be a PITA to change all the call site if it turns out to be.